refactor(core): deprecate entryComponents

crisbeto opened the pull request:
With Ivy the `entryComponents` array isn't necessary anymore. These changes mark it as deprecated so that it can be removed in a future version.

mary-poppins commented:
You can preview bf7cc34 at https://pr33205-bf7cc34.ngbuilds.io/.
You can preview 8c02a45 at https://pr33205-8c02a45.ngbuilds.io/.
You can preview 005d93d at https://pr33205-005d93d.ngbuilds.io/.

JoostK commented:
I think we should also deprecate `ANALYZE_FOR_ENTRY_COMPONENTS`.

Also, could you fix the commit message

> With Ivy the entryComponents array is **not** necessary anymore.

crisbeto commented:
The feedback has been addressed.

mary-poppins commented:
You can preview fd545ac at https://pr33205-fd545ac.ngbuilds.io/.

kara commented on aio/content/guide/deprecations.md:
```diff
@@ -351,6 +353,10 @@ As of Angular version 8, all  `platform-webworker` APIs are deprecated.
 
+{@a entryComponents}
+### `entryComponents` and `ANALYZE_FOR_ENTRY_COMPONENTS` no longer required
+Previously, the `entryComponents` array in the `NgModule` definition was used to tell the compiler, ahead of time, which components are going to be created and inserted dynamically. With Ivy this isn't a requirement anymore and the `entryComponents` array can be removed from existing module declarations. The same applies to the `ANALYZE_FOR_ENTRY_COMPONENTS` injection token.
```
```suggestion
Previously, the `entryComponents` array in the `NgModule` definition was used to tell the compiler which components would be created and inserted dynamically. With Ivy, this isn't a requirement anymore and the `entryComponents` array can be removed from existing module declarations. The same applies to the `ANALYZE_FOR_ENTRY_COMPONENTS` injection token.
```

kara reviewed (approved):
LGTM, with minor nits

crisbeto committed:
refactor(core): deprecate entryComponents

With Ivy the `entryComponents` array isn't necessary anymore. These changes mark it as deprecated so that it can be removed in a future version.

crisbeto commented:
The PR has been addressed.

mary-poppins commented:
You can preview c63a8b6 at https://pr33205-c63a8b6.ngbuilds.io/.

IgorMinar reviewed (approved):
lgtm! thanks

matsko commented:
Presubmit = http://test/OCL:275534717:BASE:275534776:1571430487778:461a47c4

Ivy Presubmit = http://test/OCL:275534717:BASE:275534836:1571430499540:b93557f4
